docs/configuration.rst

```diff
@@ -1645,7 +1645,18 @@ keys:
             The related CSS class definition must be done by the user, e.g. by :ref:`own_css`.
             (*optional*) (*default*: ``external_link``)
 
+.. _needs_import_cache_size:
 
+needs_import_cache_size
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.1.0
+
+Sets the maximum number of needs cached by the :ref:`needimport` directive,
```

Comment:
`_ImportCache` stores
`self._cache: OrderedDict[tuple[str, float], dict[str, Any]] = OrderedDict()`
where `tuple[str, float]` is path and mtime.
Items are popped until the dict size is less than `needs_import_cache_size`.
So not the number of needs is relevant, but the number of distinct needs.json paths (+mtime combination).
If I'm right, please rephrase to make that clearer.